animation.go

```diff
@@ -0,0 +1,34 @@
+package fyne
+
+import "time"
+
+// AnimationCurve represents the different animation algorithms for animation timeline.
+type AnimationCurve int
+
+const (
+	// AnimationEaseInOut is the default easing, it starts slowly, accelerates to the middle and slows to the end.
+	AnimationEaseInOut AnimationCurve = iota
```

Comment:
What ease in & out is this? There are many (Sine, Cosine, Cubic, Quadratic) so the naming should be more explicit, this maybe helpful: https://easings.net/

Reply:
I'm not certain that we need to provide all the different types, or include it in the naming. For portions of a second is it worth the technical understanding and/or the far slower maths behind the others?
If one type of EaseInOut is enough for iOS it could be good enough for us ;)

Reply:
There are many things I disagree with iOS on, so "good enough for them, good enough for us" doesn't fly with me. If there is no way for developers to set their own curves then we should provide the common ones, which don't all need to be added now, but if they will be added in future then the current naming should be explicit enough so it doesn't need to renamed when the others land.
It could be a better design to have the curve be a field of `Animation` so devs can provide their own if desired, or use a prebuilt one we provide.

Reply:
Thanks, this triggerd a thought about how this could be done better.
Update provides the same API but allows for custom animation curves - let me know what you think.